resources/benchmark-runner.mjs

```diff
@@ -178,59 +195,44 @@ class MeasureStep {
 
         this.syncTime = 0;
         this.asyncTime = 0;
         this.layoutTime = 0;
+        this.rafTime = 0;
 
         this._asyncStartTime = 0;
 
-        const label = `${suite.name}.${test.name}`;
-        this._startLabel = `${label}-start`;
-        this._syncEndLabel = `${label}-sync-end`;
-        this._asyncStartLabel = `${label}-async-start`;
-        this._asyncEndLabel = `${label}-async-end`;
-        this._forcedLayoutStartLabel = `${label}-layout-start`;
-        this._forcedLayoutEndLabel = `${label}-layout-end`;
+        this._label = `${suite.name}.${test.name}`;
+        this._startLabel = `${this._label}-start`;
+        this._syncEndLabel = `${this._label}-sync-end`;
+        this._asyncStartLabel = `${this._label}-async-start`;
+        this._asyncEndLabel = `${this._label}-async-end`;
+        this._forcedLayoutStartLabel = `${this._label}-layout-start`;
+        this._forcedLayoutEndLabel = `${this._label}-layout-end`;
+        this._rafStartLabel= `${this._label}-raf-start`;
+        this._rafEndLabel= `${this._label}-raf-end`;
+        this._rafLabel = `${this._label}-raf`;
 
         this._asyncMetricMode = params.asyncMetric;
     }
 
     async run() {
         if (this._wasRun)
             throw Error("MeasureStep can only run once.");
-        if (this._asyncMetricMode === "timeout") {
-            // 1. Schedule setTimeout measure-and-run sync work
-            // 2. setTimeout callback:
-            //    - start("sync time")
-            //    - run()
-            //    - end("sync time")
-            //    - Schedule setTimeout(..., 0)
-            //    - start("async time")
-            // 3. Potential async work
-            // 4. setTimeout callback: end("async time")
-            setTimeout(this._measureSyncCallback, 0);
-        } else if (this._asyncMetricMode === "raf") {
-            // 1. Schedule rAF callbacks:
-            //    - Schedule rAF 1: measure-and-run sync work
-            //    - Schedule rAF 2: measure async work
-            // 2. rAF 1 callback:
-            //    - start("sync time")
-            //    - run()
-            //    - end("sync time")
-            //    - start("async time")
-            // 3. rAF 2 callback: Schedule setTimeout(..., 0) measure task
-            // 4. Potential async work
-            // 5. setTimeout-callback: end("async time")
-            requestAnimationFrame(this._measureSyncCallback);
-            requestAnimationFrame(this._measureAsyncRafCallback);
-        } else {
-            throw new Error(`Unknown async measure mode: ${this._asyncMetricMode}`);
+        if (params.stepWaitTime) {
+            await new Promise((resolve) => {
+                setTimeout(resolve, params.stepWaitTime);
+            });
         }
+        if (this._asyncMetricMode === "timeout")
+            setTimeout(this._measureSync.bind(this), 0);
```

Comment:
Why don't we wrap this in callbacks class like TimerCallbacks which takes two arguments like RafBracketedCallbacks.